### PR TITLE
Add doc-test

### PR DESCRIPTION
**Pull request by MnO2**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,9 +1,59 @@
...
+//! ## Installation
+//!
+//! Add it to your `Cargo.toml`:
+//!
+//! ```toml
+//! [dependencies]
+//! darts = "0.1"
+//! ```
+//!
+//! then you are good to go. If you are using Rust 2015 you have to ``extern crate darts`` to your crate root as well.
```
Comment (andelf) [inline src/lib.rs]: s/then/Then/
... you have to **add** ...

File: src/lib.rs [hunk 2]
```diff
@@ -1,9 +1,59 @@
...
+//!
+//! ## Example
+//!
+//! ```rust
+//! use darts::DoubleArrayTrie;
+//! use std::fs::File;
```
Comment (andelf) [inline src/lib.rs:17]: should std imports come first?

File: src/lib.rs [hunk 3]
```diff
@@ -1,9 +1,59 @@
...
+//!     let string = "中华人民共和国";
+//!     let prefixes = da.common_prefix_search(string).as_ref().map(|matches| {
+//!         let res = matches
+//!             .iter()
+//!             .map(|&(end_idx, v)| {
+//!                 &string[..end_idx]
+//!             })
+//!             .collect();
+//!         res
+//!     }).unwrap_or(vec![]);
```
Comment (andelf) [inline src/lib.rs:31]: let res = foo;
    res

is equivalent to

    foo

btw, is ``into_iter`` more suitable here?

File: src/lib.rs [hunk 4]
```diff
@@ -1,9 +1,59 @@
...
+//!     assert_eq!(vec!["中", "中华", "中华人民", "中华人民共和国"], prefixes);
+//! }
+//! ```
+//!
+//! ```rust
+//! use darts::DoubleArrayTrie;
+//! use std::fs::File;
```
Comment (andelf) [inline src/lib.rs:37]: std imports comes first

File: src/lib.rs (hunk 2 above)
Comment (MnO2) [reply src/lib.rs:17]: All updated. :-)

**Pull request merged**